lib/wda-mac.js

```diff
@@ -346,6 +391,12 @@ class WDAMacServer {
         log.info(`Mac2Driver session cannot be deleted. Original error: ${e.message}`);
       }
     }
+
+    if (this.isProxyingRemoteWDAMac) {
```

Comment:
this may not work since not all clients call session delete (even though they have to). i would rather add the check into `isProxyReady` method

Reply:
ah, good point. yeah, now i think around here can simplify to check the remote server every time.